packages/docgen/lib/get-type-annotation.js

```diff
@@ -12,23 +12,40 @@ const { types: babelTypes } = require( '@babel/core' );
 /* eslint-enable jsdoc/valid-types */
 
 /**
- * @param {babelTypes.TSCallSignatureDeclaration | babelTypes.TSFunctionType} typeAnnotation
+ * @param {babelTypes.TSCallSignatureDeclaration | babelTypes.TSFunctionType | babelTypes.TSConstructSignatureDeclaration} typeAnnotation
  * @param {' => ' | ': '} returnIndicator The return indicator to use. Allows using the same function for function annotations and object call properties.
  */
 function getFunctionTypeAnnotation( typeAnnotation, returnIndicator ) {
-	const params = typeAnnotation.parameters
+	const nonRestParams = typeAnnotation.parameters
+		.filter( ( p ) => babelTypes.isIdentifier( p ) )
 		.map(
 			( p ) =>
 				`${ p.name }: ${ getTypeAnnotation(
 					p.typeAnnotation.typeAnnotation
 				) }`
 		)
 		.join( ', ' );
+
+	const {
+		argument: restParamArgument,
+		typeAnnotation: restParamTypeAnnotation,
+		// There's only every one rest param
+	} = typeAnnotation.parameters.find( babelTypes.isRestElement ) || {};
+	const restParam =
+		restParamArgument && restParamTypeAnnotation
+			? `, ...${ restParamArgument.name }: ${ getTypeAnnotation(
+					restParamTypeAnnotation.typeAnnotation
+			  ) }`
+			: '';
+
+	const params = `( ${ nonRestParams }${ restParam } )`;
```

Comment:
```js
const params = nonRestParams;
const restParam = typeAnnotation.parameters.find( babelTypes.isRestElement );
if ( restParam ) {
  const paramName = restParam.restParamArgument.name;
  const paramType = getTypeAnnotation( restParam.restParamTypeAnnotation.typeAnnotation );
  params += `, ...${ paramName }: ${ paramType }`
}


...

return `( ${ params } )${ returnIndicator }${ returnType }`
```

we chatted about this in the call a bit -- I personally think this reads more clearly. But I don't have strong opinions about changing it if it's already working fine 👍

Reply:
Oh I like that better, yes! Thank you for the suggestion.

Reply:
i guess it also assumes that if `restParam` exists, then restParamArgument and restParamTypeAnnotation will both also exist. But that seems probably true?

Reply:
Yes, that's a safe assumption.